web/src/pm_list.ts

```diff
@@ -199,16 +205,35 @@ function zoom_in(): void {
     $(".direct-messages-container").removeClass("zoom-out").addClass("zoom-in");
     $("#streams_list").hide();
     $(".left-sidebar .right-sidebar-items").hide();
+    $("#left_sidebar_scroll_container .simplebar-content-wrapper").attr("tabindex", "-1");
+
+    const $filter = $(".direct-messages-list-filter").expectOne();
+    $filter.trigger("focus");
 }
 
 function zoom_out(): void {
     zoomed = false;
-    update_private_messages();
+    if (!clear_search()) {
+        update_private_messages();
+    }
     $(".direct-messages-container").removeClass("zoom-in").addClass("zoom-out");
     $("#streams_list").show();
     $(".left-sidebar .right-sidebar-items").show();
+    $("#left_sidebar_scroll_container .simplebar-content-wrapper").attr("tabindex", "0");
+}
```

Comment:
This workaround will be removed on solving #30403 which I'll work on soon, but you can ignore this now and proceed for the rest of the review.